### ISSUE TITLE
How to create perpetual swap order on okex

### ISSUE THREAD

**Issue by Ish2K** [ccxt/ccxt#9333]
I am able to place spot order(both limit and market) using this method but unfortunately unable to make swap orders. I am able to place this order on okex dashboard and I have enough balance for order execution. What am I missing? Is there any way I can place this order using API call? Any suggestions/comments will be highly appreciated. Thank you!

- OS: Windows 10
- Programming Language version: Python 3.8
- CCXT version: 1.51.3

```
exchange_swap = ccxt.okex5({
    'apiKey': credentials['okex']['apikey'],
    'secret': credentials['okex']['secretkey'],
    'password': credentials['okex']['password'],
    'options': {
        'defaultType': 'swap',  # ←-------------- quotes and 'future'
    }
})
params =  {
    "test":True
}
order = exchange_swap.createLimitBuyOrder('XRP-USDT',100,0.7,params)
print(order)
```

```
Traceback (most recent call last):
  File "C:\Users\ishaa\Desktop\trading\Crypto Vibhor\Testing Files\ccxt_test.py", line 70, in <module>
    order = exchange_swap.createLimitBuyOrder('XRP-USDT-SWAP',100,0.7,params)
  File "C:\Users\ishaa\Desktop\trading\tradingenv\lib\site-packages\ccxt\base\exchange.py", line 1983, in create_limit_buy_order
    return self.create_order(symbol, 'limit', 'buy', amount, price, params)
  File "C:\Users\ishaa\Desktop\trading\tradingenv\lib\site-packages\ccxt\okex5.py", line 1287, in create_order
    response = self.privatePostTradeOrder(self.extend(request, params))
  File "C:\Users\ishaa\Desktop\trading\tradingenv\lib\site-packages\ccxt\base\exchange.py", line 461, in inner
    return entry(_self, **inner_kwargs)
  File "C:\Users\ishaa\Desktop\trading\tradingenv\lib\site-packages\ccxt\base\exchange.py", line 486, in request
    return self.fetch2(path, api, method, params, headers, body)
  File "C:\Users\ishaa\Desktop\trading\tradingenv\lib\site-packages\ccxt\base\exchange.py", line 482, in fetch2
    return self.fetch(request['url'], request['method'], request['headers'], request['body'])
  File "C:\Users\ishaa\Desktop\trading\tradingenv\lib\site-packages\ccxt\base\exchange.py", line 634, in fetch
    self.handle_errors(http_status_code, http_status_text, url, method, headers, http_response, json_response, request_headers, request_body)
  File "C:\Users\ishaa\Desktop\trading\tradingenv\lib\site-packages\ccxt\okex5.py", line 2230, in handle_errors
    self.throw_exactly_matched_exception(self.exceptions['exact'], errorCode, feedback)
  File "C:\Users\ishaa\Desktop\trading\tradingenv\lib\site-packages\ccxt\base\exchange.py", line 500, in throw_exactly_matched_exception
    raise exact[string](message)
ccxt.base.errors.BadRequest: okex5 {"code":"1","data":[{"clOrdId":"","ordId":"","sCode":"51000","sMsg":"Parameter posSide  error ","tag":""}],"msg":""}
```

**Comment by kroitor** [comment]
```Python
exchange_swap = ccxt.okex5({
    'apiKey': credentials['okex']['apikey'],
    'secret': credentials['okex']['secretkey'],
    'password': credentials['okex']['password'],
    'options': {
        'defaultType': 'swap',
    }
})
params =  {
    # "test":True,  # ←------------ THIS LINE SHOULD NOT BE HERE
}
markets = exchange_swap.load_markets()
# exchange_swap.verbose = True  # uncomment for debugging purposes
order = exchange_swap.createLimitBuyOrder('XRP-USDT-SWAP', 100, 0.7)
print(order)
```

Let us know if the above does not answer your question. If you still have issues with it, please, paste your complete verbose request+response: https://github.com/ccxt/ccxt/wiki/FAQ#what-is-required-to-get-help

**Comment by Ish2K** [comment]
Hey! Thanks for the reply! 
I solved the issue, the solution was to define 'posSide'  (long,short) and also change 'tdMode' to isolated in parameters. 
order = exchange_swap.createLimitBuyOrder('XRP-USDT-SWAP', 100, 0.7)
In this line '100' is actually the lot size (which in this case is 100), so I was placing order of 100*100 XRP (amount). 
After making the above changes I was able to successfully place the order.
